src/main/java/com/piekoszek/nowaksharedrent/auth/AuthServiceImpl.java

```diff
@@ -42,8 +45,20 @@ private String returnToken (Account account) {
         Account registeredAccount = accountRepository.findByEmail(account.getEmail());
         if (registeredAccount != null && hashService.compareWithHash(account.getPassword(), registeredAccount.getPassword()))
         {
-            return Optional.of(returnToken(registeredAccount));
+            return Optional.of(jwtService.generateToken(userRepository.findByEmail(account.getEmail())));
```

Comment:
I guess there is possible null pointer exception if findByEmail(account.getEmail()) return null, but shouldn't happen, so I'd leave it as it is, cos InternalServerError is good response in this case.